client/client.lua

```diff
@@ -1004,183 +602,644 @@ local function OpenBanMenu(banplayer)
+            description = Lang:t("weather.clearing_desc")
+        },
+        [9] = {
+            icon = '☂️',
+            label = Lang:t("weather.rain"),
+            value = "RAIN",
+            description = Lang:t("weather.rain_desc")
+        },
+
+        [10] = {
+            icon = '⛈️',
+            label = Lang:t("weather.thunder"),
+            value = "THUNDER",
+            description = Lang:t("weather.thunder_desc")
+        },
+        [11] = {
+            icon = '❄️',
+            label = Lang:t("weather.snow"),
+            value = "SNOW",
+            description = Lang:t("weather.snow_desc")
+        },
+        [12] = {
+            icon = '🌨️',
+            label = Lang:t("weather.blizzard"),
+            value = "BLIZZARD",
+            description = Lang:t("weather.blizzed_desc")
+        },
+        [13] = {
+            icon = '❄️',
+            label = Lang:t("weather.light_snow"),
+            value = "SNOWLIGHT",
+            description = Lang:t("weather.light_snow_desc")
+        },
+        [14] = {
+            icon = '🌨️',
+            label = Lang:t("weather.heavy_snow"),
+            value = "XMAS",
+            description = Lang:t("weather.heavy_snow_desc")
+        },
+        [15] = {
+            icon = '🎃',
+            label = Lang:t("weather.halloween"),
+            value = "HALLOWEEN",
+            description = Lang:t("weather.halloween_desc")
+        }
+    }
+    for _, v in ipairs(elements) do
+        menu8:AddButton({icon = v.icon,label = v.label,value = v,description = v.description,select = function(btn)
+            local selection = btn.Value
+            TriggerServerEvent('qb-weathersync:server:setWeather', selection.value)
+            QBCore.Functions.Notify(Lang:t("weather.weather_changed", {value = selection.label}))
+        end})
+    end
+end)
+
+-- Adjust time on change
+menu3_server_time:On("select", function(_, value)
+    TriggerServerEvent("qb-weathersync:server:setTime", value, value)
+    QBCore.Functions.Notify(Lang:t("time.changed", {time = value}))
+end)
+
+--[[
+    Vehicle Spawner functions
+--]]
+
+-- Set vehicle Categories
+local vehicles = {}
+for k, v in pairs(QBCore.Shared.Vehicles) do
+    local category = v["category"]
+    if vehicles[category] == nil then
+        vehicles[category] = { }
+    end
+    vehicles[category][k] = v
+end
+
+-- Car Categories
+local function OpenCarModelsMenu(category)
+    menu13:ClearItems()
+    MenuV:OpenMenu(menu13)
+    for k, v in pairs(category) do
+        menu13:AddButton({
+            label = v["name"],
+            value = k,
+            description = 'Spawn ' .. v["name"],
+            select = function(_)
+                TriggerServerEvent('QBCore:CallCommand', "car", { k })
+            end
+        })
+    end
+end
+
+menu5_vehicles_spawn:On('Select', function(_)
+    menu12:ClearItems()
+    for k, v in pairs(vehicles) do
+        menu12:AddButton({
+            label = QBCore.Shared.FirstToUpper(k),
```

Comment:
The issue of not seeing the vehicle categories will be because of this line here. I created a PR in qb-core to address this which can be found [here](https://github.com/qbcore-framework/qb-core/pull/809) @tom-osborne.

The Dealer List not showing any items is due to no one dealing I believe. Though I have not thoroughly tested this, it should work as it previously has as no functions were inherently changed in this PR.